### ISSUE TITLE
Experiments cannot run with eval_every == 0

### ISSUE THREAD

**Issue by guggio** [deepset-ai/FARM#307]
**Describe the bug**
For my experiment, I don't use a dev set and thus don't need the dev evaluation after a certain amount of training steps. Therefore, I set the eval_every variable to 0 in my experiment-json (as suggested in the description in the json) . Unfortunately, this leads to a ZeroDivisionError as listed below. It seems to me, an additional conditional statement (self.evaluate_every != 0) in line 249 of train.py would resolve this issue. I was able to perform my experiment by setting it to a number, larger than the total training steps, but this is just fighting the symptoms. 

Thanks guys and keep up the great work!  

**Error message**

```
/usr/local/lib/python3.6/dist-packages/farm/train.py in train(self)
    247 
    248                 # Perform  evaluation
--> 249                 if self.global_step % self.evaluate_every == 0 and self.global_step != 0:
    250                     # When using StreamingDataSilo, each evaluation creates a new instance of
    251                     # dev_data_loader. In cases like training from scratch, this could cause

ZeroDivisionError: integer division or modulo by zero
```

**Expected behavior**
No Error :)

**To Reproduce**
Run an experiment with the eval_every variable set to 0 (value or default, doesn't matter).

**System:**
 - OS: macOS 10.15
 - GPU/CPU: 
 - FARM version: latest master

**Comment by Timoeller** [comment]
Hey @guggio thanks for reporting this bug. It looks like we need a simple check.

Would you be interested in adding the check and creating a PR since you spotted the problem? I can also gladly take over.

**Comment by Timoeller** [comment]
Fixed with #311 - closing this.
